pkg/controller/eventing/subscription/reconcile.go

```diff
@@ -0,0 +1,331 @@
+	resourceClient, err := r.CreateResourceInterface(namespace, ref)
+	if err != nil {
+		glog.Warningf("failed to create dynamic client resource: %v", err)
+		return nil, err
+	}
+
+	return resourceClient.Get(ref.Name, metav1.GetOptions{})
+}
+
+func (r *reconciler) reconcileFromChannel(namespace string, subscribable corev1.ObjectReference, callDomain string, resultDomain string, deleted bool) error {
+	glog.Infof("Reconciling From Channel: %+v call: %q result %q deleted: %v", subscribable, callDomain, resultDomain, deleted)
+
+	s, err := r.fetchObjectReference(namespace, &subscribable)
+	if err != nil {
+		return err
+	}
+
+	c := duckv1alpha1.Channel{}
+	err = duck.FromUnstructured(*s, &c)
+	if err != nil {
+		return err
+	}
+
+	glog.Infof("Found From Channelable: %+v", c)
+
+	// Add our subscription to the object
+	after := c.DeepCopyObject().(*duckv1alpha1.Channel)
+	after.Spec.Channelable = &duckv1alpha1.Channelable{
+		Subscribers: []duckv1alpha1.ChannelSubscriberSpec{{CallableDomain: "foobar"}},
+	}
+
+	patch, err := createPatch(c, after)
+	if err != nil {
+		return err
+	}
+	glog.Infof("PATCH is: %+v", patch)
+
+	patchBytes, err := patch[0].MarshalJSON()
+	if err != nil {
+		glog.Warningf("failed to marshal json patch: %s", err)
+		return err
+	}
+
+	resourceClient, err := r.CreateResourceInterface(namespace, &subscribable)
+	if err != nil {
+		glog.Warningf("failed to create dynamic client resource: %v", err)
+		return err
+	}
+	patched, err := resourceClient.Patch(c.Name, types.JSONPatchType, patchBytes)
+	if err != nil {
+		glog.Warningf("Failed to patch the object: %s", err)
+		return err
+	}
+	glog.Warningf("Patched resource: %+v", patched)
+	return nil
+}
+
+func (r *reconciler) CreateResourceInterface(namespace string, ref *corev1.ObjectReference) (dynamic.ResourceInterface, error) {
+	gvk := ref.GroupVersionKind()
+
+	rc := r.dynamicClient.Resource(schema.GroupVersionResource{
+		Group:    gvk.Group,
+		Version:  gvk.Version,
+		Resource: pluralizeKind(gvk.Kind),
+	})
+	if rc == nil {
+		return nil, fmt.Errorf("failed to create dynamic client resource")
+	}
+	return rc.Namespace(namespace), nil
+
+}
+
+// takes a kind and pluralizes it. This is super terrible, but I am
+// not aware of a generic way to do this.
+// I am not alone in thinking this and I haven't found a better solution:
+// This seems relevant:
+// https://github.com/kubernetes/kubernetes/issues/18622
+func pluralizeKind(kind string) string {
+	ret := strings.ToLower(kind)
+	if strings.HasSuffix(ret, "s") {
+		return fmt.Sprintf("%ses", ret)
+	}
+	return fmt.Sprintf("%ss", ret)
+}
+
+func createPatch(before, after interface{}) ([]jsonpatch.JsonPatchOperation, error) {
+	// Marshal the before and after.
+	rawBefore, err := json.Marshal(before)
+	if err != nil {
+		return nil, err
+	}
+
+	rawAfter, err := json.Marshal(after)
+	if err != nil {
+		return nil, err
+	}
+
+	return jsonpatch.CreatePatch(rawBefore, rawAfter)
+}
+
+const (
```

Comment:
I think all of these constants could be deleted - they don't seem to be used anywhere.

Reply:
done.